modules/300-prometheus/hooks/deprecate_outdated_grafana_dashboard_crd.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/tidwall/gjson"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/deckhouse/deckhouse/go_lib/dependency"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/prometheus/deprecate_outdated_grafana_dashboard_crd",
+	Schedule: []go_hook.ScheduleConfig{
+		{
+			Name:    "helm_releases",
+			Crontab: "0 * * * *", // every hour
+		},
+	},
+}, dependency.WithExternalDependencies(handleGrafanaDashboardCRDs))
+
+func handleGrafanaDashboardCRDs(input *go_hook.HookInput, dc dependency.Container) error {
```

Comment:
https://github.com/deckhouse/deckhouse/pull/7027#discussion_r1465293281

> Deckhouse does not subscribe to Grafana dashboard definitions and helm release secrets because they are too heavy to keep in memory.
Since this is only for the deprecation, consider switching to the schedule binding. Example: https://github.com/deckhouse/deckhouse/blob/main/modules/340-monitoring-kubernetes/hooks/helm.go

> Switched to use the schedule